src/main/java/org/junit/internal/ComparisonCriteria.java

```diff
@@ -56,27 +68,31 @@ public void arrayEquals(String message, Object expecteds, Object actuals)
                 }
             }
         }
-    }
 
-    private boolean isArray(Object expected) {
-        return expected != null && expected.getClass().isArray();
+        if (actualsLength != expectedsLength) {
+            Object expected = getArrayElementOrSentinel(expecteds, expectedsLength, prefixLength);
+            Object actual = getArrayElementOrSentinel(actuals, actualsLength, prefixLength);
+            try {
+                Assert.assertEquals(expected, actual);
```

Comment:
The previous code just used `Assert.fail()` here. Any reason we should do something different? I seen to recall this code being a bit sensitive when handling nested arrays.